core/src/main/java/org/ehcache/core/resilience/RobustLoaderWriterResilienceStrategy.java

```diff
@@ -64,12 +66,11 @@ public boolean containsKeyFailure(K key, StoreAccessException e) {
   @Override
   public void putFailure(K key, V value, StoreAccessException e) {
     cleanup(key, e);
-  }
-
-  @Override
-  public void putFailure(K key, V value, StoreAccessException e, CacheWritingException f) {
-    cleanup(key, e);
-    throw f;
+    try {
+      loaderWriter.write(key, value);
```

Comment:
I'm thinking this should write then cleanup.  With this ordering we could cleanup, have another thread populate through the loader and then write.  That leaves the cache with the wrong contents, right?

Reply:
Yes